cloudsync/tasks_test.py

```diff
@@ -229,3 +236,106 @@ def test_stream_to_s3_no_video():
+    bucket.upload_fileobj(io.BytesIO(os.urandom(6250000)), upload)
+    assert s3c.get_object(Bucket=bucket.name, Key=upload) is not None
+    monitor_watch_bucket.delay()
+    new_video = Video.objects.get(title=upload)
+    new_videofile = new_video.videofile_set.get(encoding=EncodingNames.ORIGINAL)
+    assert new_videofile.bucket_name == settings.VIDEO_S3_BUCKET
+    with pytest.raises(ClientError):
+        s3c.get_object(Bucket=bucket.name, Key=upload)
+
+
+@pytest.mark.django_db
+@mock_s3
+@override_settings(LECTURE_CAPTURE_USER='admin')
+def test_monitor_watch_exceptions(mocker, user):  # pylint: disable=unused-argument,redefined-outer-name
+    """
+    Test that an exception is correctly handled
+    """
+    s3 = boto3.resource('s3')
+    s3c = boto3.client('s3')
+    upload = 'MIT-6.046-2017-Spring-lec-mit-0000-2017apr06-0404-L01.mp4'
+    s3c.create_bucket(Bucket=settings.VIDEO_S3_WATCH_BUCKET)
+    bucket = s3.Bucket(settings.VIDEO_S3_WATCH_BUCKET)
+    bucket.upload_fileobj(io.BytesIO(os.urandom(6250000)), upload)
+    with pytest.raises(User.DoesNotExist) as exc:
+        monitor_watch_bucket.delay()
+        assert 'User matching query does not exist' in str(exc)
+    user.username = 'admin'
+    user.save()
+    with pytest.raises(ClientError) as exc:
+        monitor_watch_bucket.delay()
+    assert 'NoSuchBucket' in str(exc)
+    mocker.patch('cloudsync.tasks.process_watch_file',
+                 side_effect=ValueError('Fake ValueError'))
+    with pytest.raises(ValueError) as exc:
+        monitor_watch_bucket.delay()
+    assert 'Fake ValueError' in str(exc)
+
+
+def test_extract_title():
+    """
+    Test that a correctly formatted title is returned for a video file.
+    """
+    filenames = ('MIT-6.046-2017-Spring-lec-mit-0000-2017apr06-0404-L01.mp4',
+                 'MIT-6.046-2018-Fall-lec-mit-0000-2017apr06-0404.mp4',
+                 'MIT-6.046-lec-mit-0000-2017apr06-0404.mp4',
+                 'MIT-6.046-lec-mit-0000-2017apr06-0404-L02.mp4',
+                 'MIT-STS.001-2015-Fall-lec-mit-0000-2015dec21-2222-L03_2.mp4',
+                 'MIT-HST.969-2018-Spring-lec-mit-0000-2018mar09-0001-L01.mp4',
+                 'NUS-7.S939-2019-Spring-lec-mit-0000-2019apr02-1210-L05.mp4',
+                 'MIT-Adhoc-lec-mit-0000-2017aug10-0954-2-190.mp4',
+                 'MIT-18.test-lec-mit-0000-2017aug10-0956-L01.mp4',
+                 'MIT-18.test-lec-mit-0000-2017aug10-0956.mp4',
+                 'MIT-Simons Lecture Series-lec-mit-0000-2017aug10-0956-2-10.mp4')
+    expected = ('MIT-6.046-2017-Spring-L01',
+                'MIT-6.046-2018-Fall',
+                'MIT-6.046',
+                'MIT-6.046-L02',
+                'MIT-STS.001-2015-Fall-L03',
+                'MIT-HST.969-2018-Spring-L01',
+                'NUS-7.S939-2019-Spring-L05',
+                'MIT-Adhoc-2-190',
+                'MIT-18.test-L01',
+                'MIT-18.test')
+    for item in zip(filenames, expected):
```

Comment:
I think that is better if you put together the two list in tuples where each tuple is input and output. This is already difficult to read.